llvm/lib/CodeGen/SelectionDAG/LegalizeVectorOps.cpp

```diff
@@ -1026,6 +1026,13 @@ void VectorLegalizer::Expand(SDNode *Node, SmallVectorImpl<SDValue> &Results) {
       return;
     }
     break;
+  case ISD::VP_CTTZ_ELTS:
```

Comment:
I think this should actually be expanded in LegalizeDAG. The original intent of LegalizeVectorOps is to handle instructions that may need to scalarize. Anything that is exclusively vector with no scalar equivalent should be done by LegalizeDAG.

Unfortunately, the code for VP intrinsics in `VectorLegalizer::LegalizeOp` where we instantiate BEGIN_REGISTER_VP_SDNODE is completely unaware of this. I think we should force `Legal` for this operation there so that it will get ignored until LegalizeDAG.

Reply:
Done.